Draft - Diff with the community to look at

DonRichards opened the pull request.

emetsger committed:
Add GNU patch for use with composer-patches.

dannylamb committed:
Using secrets

nigelgbanks committed:
Merge pull request #79 from jhu-sheridan-libraries/gnu-patch

Add GNU patch for use with composer-patches.

dannylamb committed:
Update allowed-external-content.txt.tmpl

dannylamb committed:
Missing key for /tomcat/admin/roles

nigelgbanks committed:
Merge pull request #81 from Islandora-Devops/allowed-external-content-fix

Update allowed-external-content.txt.tmpl

nigelgbanks committed:
Merge pull request #82 from Islandora-Devops/missing-tomcat-admin-roles

Missing key for /tomcat/admin/roles

dannylamb committed:
Merge branch 'main' of https://github.com/Islandora-Devops/isle-buildkit into secrets

dannylamb committed:
Guess this is no longer neccessary to build

emetsger committed:
Use the output of git-describe to tag images when the version property is absent or the empty string.

nigelgbanks committed:
Merge pull request #83 from emetsger/git-describe-version

Version Docker images using `git describe` when no `version` is specified

dannylamb committed:
Update Dockerfile

dannylamb committed:
Typo

dannylamb committed:
Merge branch 'secrets' of https://github.com/Islandora-Devops/isle-buildkit into secrets

nigelgbanks committed:
Allow drupal to lock tables, allows importing dumps generated by Drush: https://www.drupal.org/project/drupal/issues/2487089. (#86)

Commit:
Merge pull request #80 from Islandora-Devops/secrets

Using secrets

dannylamb committed:
Injecting DRUPAL_REVERSE_PROXY_IP for traefik if it's available

nigelgbanks committed:
Allow DRUPAL_REVERSE_PROXY_IPS to be set explicitly via whatever backend is configured for confd. (#88)

nigelgbanks committed:
Merge pull request #87 from Islandora-Devops/sprint4

Injecting DRUPAL_REVERSE_PROXY_IP for traefik if it's available

nigelgbanks committed:
Composer 2 (#85)

* Allow drush to be called as non-root user.

* Upgrade to composer version 2.

nigelgbanks committed:
Allow for multiple php-fpm services to be run in downstream containers. (#84)

emudojo committed:
Merge pull request #1 from Islandora-Devops/main

Updating from upstream

Commit:
Adding default config dir

Commit:
Addinf Suppport for default configuration

Commit:
added nginx configuration changes for the vhost

nigelgbanks committed:
Merge pull request #94 from Born-Digital-US/adding-config-directory-demo

Adding proxybuffer updates to prevent 502 errors, also enable SSL by default always

dannylamb committed:
Sandbox, Import/Export tool, and Syn toggling

nigelgbanks committed:
Have imagemagick depend on ghostscript so PDF thumbnail generation works more broadly (#93)

nigelgbanks committed:
Automatically build/publish to DockerHub using the git tag for the docker image when a new release is minted.

dannylamb committed:
Changes for reindexing and no gemini

Tristaan committed:
Handle.net persistent identifiers

This commit adds the handle.net server as a service to isle-buildkit.

dannylamb committed:
Enabling buildkit

dannylamb committed:
Testing out every commit

dannylamb committed:
Making it just commit and dropping branch name

nigelgbanks committed:
Merge pull request #99 from nigelgbanks/release-versioning

Automatically build / publish to DockerHub

nigelgbanks committed:
Clear git repository cache and re-attempt if checkout for given commit fails, clear cache on script failure. (#97)

dannylamb committed:
Getting rid of unneccessary cut

nigelgbanks committed:
Merge pull request #105 from dannylamb/release-versioning

Release versioning

dannylamb committed:
Adding back action to build PRs.  Removing unneccessary action to build and push latest

dannylamb committed:
Trying to fix build fail

nigelgbanks committed:
Merge pull request #106 from dannylamb/release-versioning

Adding back action to build PRs.  Removing unneccessary action to bui…

nigelgbanks committed:
Added missing docs for the public/private keys

nigelgbanks committed:
Made logic for creating a service user generic so it could be used by things other than apache tools.

nigelgbanks committed:
Updated handle changes to prioritize configuration from confd rather than from environment. Updated docs, and fixed issues found during testing.

dannylamb committed:
Undoing some wrong things i did

dannylamb committed:
Merge branch 'main' of https://github.com/islandora-devops/isle-buildkit into post-sprint-4

dannylamb committed:
Facepalming on no /rootfs

dannylamb committed:
Updating islandora-sandbox

dannylamb committed:
Updating islandora-sandbox

dannylamb committed:
Fedora 6!

dannylamb committed:
Updating utilities for local test case

dannylamb committed:
Remote activemq with fcrepo6.  Reworking the fedoraAdmin permission during init scripts

dannylamb committed:
Re-triggering github action to push new images

Tristaan committed:
Add additional env vars to handle service

nigelgbanks committed:
Matomo support multisites (#78)

* More stability in building.

* Multisite matomo

* Make it work with MySQL as well as MariaDB

* Have Matomo work with/without traefik prefix strip (ALB does not support stripping paths).

nigelgbanks committed:
Use explicit frontend version of buildkit, should help with some downstream bugs. (#110)

nigelgbanks committed:
Refactor how environment variables secrets and defaults are set. (#96)

* Refactor how environment variables secrets and defaults are set.

* Bump build time for the occasions where a full rebuild is required.

nigelgbanks committed:
Try multiple keyservers for fetching gpg keys to increase build stability. (#109)

Tristaan committed:
Merge branch 'main' of github.com:Abak-net-d-o-o/isle-buildkit into main

nigelgbanks committed:
Changed to follow new conventions around Confd/Environment variables.

nigelgbanks committed:
Removed reference to Etcd, as any Confd backend might be used

nigelgbanks committed:
Fixed bug in postgres startup script causing failure to start.

nigelgbanks committed:
Fix permission issue with PostGres that prevented startup.

nigelgbanks committed:
Bumped version of gradle docker plugin and enabled caching (#108)

Tristaan committed:
Merge pull request #1 from nigelgbanks/handle-branch-updates

Handle branch updates

nigelgbanks committed:
Merge pull request #111 from Abak-net-d-o-o/main

Handle.net persistent identifiers service

dannylamb committed:
Merging in upstream changes

dannylamb committed:
Adding in getenvs

dannylamb committed:
Rebasing off of nigel's work

noahwsmith committed:
Remove Gemini from list of require services

dannylamb committed:
Missing FCREPO_DB_ROOT_PASSWORD

dannylamb committed:
Reworking how I check if modules are enabled or not

nigelgbanks committed:
Temporary fix for Kubernetes

nigelgbanks committed:
Merge pull request #114 from nigelgbanks/kubernetes-support

Temporary fix for Kubernetes

nigelgbanks committed:
Merge branch 'main' into post-sprint-4

Tristaan committed:
Merge pull request #2 from Islandora-Devops/main

Pull changes from islandora

Tristaan committed:
Change mysql root user creation order

When the grant happens after set password the password is not working.
So the grant needs to happen before setting the password.

nigelgbanks committed:
Merge pull request #116 from Abak-net-d-o-o/main

Fix mariadb root@% password issue

dannylamb committed:
Updating to use newest code. Touching up recast

dannylamb committed:
Merge branch 'post-sprint-4' of https://github.com/islandora-devops/isle-buildkit into post-sprint-4

dannylamb committed:
Updating crayfish commit

dannylamb committed:
Responding to feedback.  README updates

dannylamb committed:
Touching up utilities.sh

dannylamb committed:
Attempting to fix deploying on remote instance without traefik.me

DonRichards committed:
Update README.md

https://github.com/fits4/fits4

nigelgbanks committed:
Merge pull request #120 from Islandora-Devops/DonRichards-patch-1

Update README.md

nigelgbanks committed:
Merge pull request #95 from Islandora-Devops/post-sprint-4

Sandbox, Import/Export tool, and Syn toggling

nigelgbanks committed:
Use the version 0.0.4 of the build plugin

Drops support for non-buildkit builds. Adds the following:

- multi-arch builds (amd64, arm64, etc)
- custom build of confd to support arm64
- unique build caches per image (should reduce cache misses)
- fixes fits (mediainfo, exifdata, file info)
- update gradle version to latest (match plugin)
- child images of tomcat can override tomcat properties
- child images of nginx can override php/nginx properties
- updated to use latest alpine 3.13
- php is no 7.4
- imagemagick only builds what is required (no longer make clibs, etc)
- general refactors

nigelgbanks committed:
Running out of disk space on Github build servers

When not pushing to remote repository so use the default builder instead
for pull requests.

nigelgbanks committed:
Refactor of environment variable usage (#122)

- Removed prefixes from each image confd settings.
- Allows the use/overriding of settings defined by parent images.
- Shared database/jwt configuration.
- Conslidated common behavoir for importing/overriding variables.
- Conslidated common behavoir for database creation
- Updates to documentation
- Removed references to GEMINI

nigelgbanks committed:
Fix for permissions issues in tomcat.

nigelgbanks committed:
Fix to use the right environment variables for database root user.

nigelgbanks committed:
Automated Tests

- Added automated tests and documentation.
- Return the exit code of the service that exited
- Previously always returned 0 even if the service failed to start.

nigelgbanks committed:
Bump version of plugin

dannylamb committed:
Pulling in latest islandora code to fix chullo warning

nigelgbanks committed:
tomcat/nginx config folder location change

nigelgbanks committed:
Update nginx config file to match most recent version from updated Alpine

nigelgbanks committed:
Update php-fpm config files to match most recent version from updated Alpine

nigelgbanks committed:
Include patch in download image. (#135)

nigelgbanks committed:
Add Riprap as a microsevice (#113)

* Automated Tests

- Added automated tests and documentation.
- Return the exit code of the service that exited
- Previously always returned 0 even if the service failed to start.

* Bump version of plugin

* Added Riprap micro-service.

nigelgbanks committed:
Coder (#130)

* Added code-server image for local/remote development

* Added phpcpd

nigelgbanks committed:
Upgrade composer from 2.0.4 to 2.0.12 (#137)

Github has release a new format for personal access tokens.

https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/

Causing a failure to authenticate bugs:

https://github.com/composer/composer/issues/9820

Which was fixed in release 2.0.12:

https://github.com/composer/composer/commit/dc83ba93f3d8a35629f9a387632e8cd373a144d0

noahwsmith committed:
Adds php7-intl (#142)

* Adds php7-intl

Addresses #141

* Also requires php7-zip

dannylamb committed:
Updating sandbox commit (#140)

nigelgbanks committed:
Remove flakey test (#138)

See for more info https://github.com/Islandora-Devops/isle-buildkit/issues/136

nigelgbanks committed:
Fix illegal instruction issue in confd (arm64 only). (#139)

dannylamb committed:
Bumping sandbox commit

emudojo committed:
Enable s3 for f6 (#134)

* Accept parameters for s3 storage

* Documenting available values for S3

* Fix formatting in documentation

* Added default s3 options and AWS credentials

* removed test.

* Updated flags to match what fedora is actually expecting

* Removed overrride for AWS_ keys and added a negative timepout for now

Co-authored-by: Noah W. Smith <noah@born-digital.com>
Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com>
Co-authored-by: Hertzel Armengol <hertzel@born-digital.com>

nigelgbanks committed:
Required dependencies for logging for `docker exec` shell sessions in cloud watch (#146)

dannylamb committed:
Bumping sandbox commit

dannylamb committed:
Bumping sandbox commit

elizoller committed:
upgrade fedora 6 to release candidate 2 (#147)

dannylamb committed:
Updating SHA256 for fcrepo6 rc 2 (#148)

nigelgbanks committed:
Remove GPG checks as they occasionally fail to contact the keyserver, and we already verify the download by its checksum. (#149)

emetsger committed:
correct fits-service.properties typo in confd toml. (#151)

dannylamb committed:
Correcting newlines on template for allowed external content in fcrepo (#150)

dannylamb committed:
Fixing erroneous config variable name and also adding the ngix user to the jwt group on hypercube (#152)

nigelgbanks committed:
Updated plugin / gradle to latest (#153)

* Updated plugin/gradle to latest

Can now perform build scans with `--scan` argument.

* Fix circular dependency in test

nigelgbanks committed:
Allow txt/log files to be served from flysystem. (#155)

dannylamb committed:
Updating to fcrepo 6.0.0

dannylamb committed:
Bumping Alpaca to 1.0.5

dannylamb committed:
Adding configuration for Alpaca timeouts

nigelgbanks committed:
Bumped Crayfish to 2.0.0 release (#160)

ruebot committed:
Remove file.stream references. (#157)

- Align with
  - https://github.com/Islandora/Alpaca/pull/78
  - https://github.com/Islandora/Alpaca/releases/tag/Alpaca-1.0.5
  - https://github.com/Islandora-Devops/isle-buildkit/blob/b3e3f46545e94a25ffe18613e63d2b984621cd7e/alpaca/Dockerfile#L8

nigelgbanks committed:
Automatically update DockerHub iage documentation. (#163)

nigelgbanks committed:
Added smaller README (<25K) for cantaloupe so it can be used with DockerHub

nigelgbanks committed:
Fix relative path for DockerHub docs

nigelgbanks committed:
Fix logging errors with cantaloupe due to find/replace bug (#165)

dannylamb committed:
Adding max_input_vars to php.ini (#166)

rosiel committed:
Explain how Docker Hub factors in the workflow. (#171)

nigelgbanks committed:
Updated the introduction to redirect users to better starting points in case they ended up at this repo by accident. (#164)

nigelgbanks committed:
[issue-172] Use explicit commits in URLs such that (#176)

Code can change when pulling from a master branch, which causes checksum
checks to fail.

This can changes master to instead refer to explicit commits hashes.

The output docker images is byte for byte the same though.

dannylamb committed:
Adding a custom delegate to Cantaloupe to pass along the Authorization header (#175)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com>

jasonhildebrand committed:
Refactor how we set/restore ownership of drupal settings file to allow reuse... (#179)

* Refactor how we set/restore ownership of drupal settings file to allow reuse in containers based on this one.

* Clarify comment.

* Adjustments.

nigelgbanks committed:
Added LDAP extension to nginx/php image. (#181)

jasonhildebrand committed:
If SOLR config.zip cannot be produced, show useful error message. (#182)

nigelgbanks committed:
[issue-190] Ensure the directory is writable before copying default.settings.php to settings.php (#191)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Remove unnecessary reverse proxy. (#187)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

dannylamb committed:
Timeouts (#185)

* _all_ the timeouts

* Missed a file

* Moved the nginx timeouts to the http context so they would apply everywhere as defaults (#188)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

Co-authored-by: Nigel Banks <nigelgbanks@users.noreply.github.com>
Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

dannylamb committed:
Missed a tomcat connection timeout when auditing for timeouts earlier. (#195)

dannylamb committed:
Update default 3-store URL now that we've removed the nginx layer (#196)

nigelgbanks committed:
Use lock file for installation of code-server (#198)

* Use lock files to pin versions of dependencies

For both:

- code-server
- composer

* Updated code-server extensions.

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

Commit:
1st pass at release process and maintainers docs (#199)

nigelgbanks committed:
Change fits default port after proxy removal; addresses #201 (#202)

Co-authored-by: Jeffery Antoniuk <jeffery.antoniuk@ualberta.ca>

nigelgbanks committed:
Issue 168 vulnerability checks (#183)

* Update plugin to latest to get vunerability analysis checks

* Bumped alpine version to latest stable.

* Updated Imagemagick to latest

* Fix for java-jna package discontinued for aarch64

* Added link to aports issue that requires custom build of java-jna

* Generate grype reports on pushs as well as pull requests.

* Updated fits to latest to get security updates.

* Added ability to ignore specific vulnerabilities, while a fix is not availiable.

* Updated tomcat to latest

* Documented additional flag for grype

* Made alpine version a argument.

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Bump timeout to 360 as a full cross platform rebuild with tests takes… (#200)

* cleanup formatting

* Bump timeout to 360 as a full cross platform rebuild with tests takes more than 60 minutes now.

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Roll back FITS to working version. Upgrade when https://github.com/harvard-lts/fits/issues/256 is included in the next release. (#208)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Demo image to use islandora install profile (#209)

Co-authored-by: Rosie Le Faive <lefaive@gmail.com>
Co-authored-by: dannylamb <dlamb@islandora.ca>

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>
Co-authored-by: Rosie Le Faive <lefaive@gmail.com>
Co-authored-by: dannylamb <dlamb@islandora.ca>

nigelgbanks committed:
Add dependency for gs back into imagemagick (#211)

nigelgbanks committed:
Issue 174 - Logging (#206)

Commit:
Adding testing isle-buildkit doc (#212)

noahwsmith committed:
migrate:import no longer supports --group function (#214)

nigelgbanks committed:
No longer use volume for Matomo install (#143)

* No longer use volume for Matomo install

Having the code base in a volume makes upgrades and migrations a pain,
additionaly the file copies on initial startup can be really slow on NFS
drives.

Having a volume allows for the dynamic installation of plugins, but then
that is specific to that instance, if you were to run a several stage
environment dev/stage/prod then one would need to manually check that
the plugins exist is each environment. Better would be to extend the
image and pack the plugins in with the installation.

@see https://github.com/matomo-org/docker/issues/57
@see https://github.com/matomo-org/docker/issues/161

* Matomo: create a per-site administrator in addition to the super administrator for all sites.

misilot committed:
Update ALPACA_FCREPO_URL value to port 8080 (#222)

Fixes #221

misilot committed:
changed sentence structure in Karaf Readme (#223)

The convention is listed below, and not above.

nigelgbanks committed:
Expose JMS configuration as environment variables. (#226)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Allow for the use of development certifications for inter-container communication. (#227)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Nginx drupal rule for protect files and directories was missing leading slash for path. (#228)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Respect X-Forwarded headers in responses. (#229)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Missing x-forwarded-port (#230)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Imagemagick url changed location required to be able to build now. (#231)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

nigelgbanks committed:
Allow development certificates to be provided as secrets rather than (#232)

mounts.

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

noahwsmith committed:
Upgrade FCREPO to 6.2.0 (#233)

Commit:
All debug log levels changed to info. Less talkative log output per issue #217 (#236)

Commit:
Upgrade to FITS 1.5.5 (#234)

nigelgbanks committed:
Upgrade Gradle/Plugin to support running tests on M1 macs (#235)

dependabot[bot] committed:
Bump vm2 in /code-server/rootfs/usr/local/share/.config/yarn/global (#247)

Bumps [vm2](https://github.com/patriksimek/vm2) from 3.9.9 to 3.9.11.
- [Release notes](https://github.com/patriksimek/vm2/releases)
- [Changelog](https://github.com/patriksimek/vm2/blob/master/CHANGELOG.md)
- [Commits](https://github.com/patriksimek/vm2/compare/3.9.9...3.9.11)

---
updated-dependencies:
- dependency-name: vm2
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nigelgbanks committed:
Fix for failing tests

nigelgbanks committed:
fix fits logging (#2) (#251)

Co-authored-by: Seth Shaw <108362375+seth-shaw-asu@users.noreply.github.com>

Commit:
Issue 224 Fixed - Dead code removed (#241)

* Issue 224 Fixed - Dead code removed

* Issue 224 Fixed - Dead code removed and Milliner README fixed

dannylamb committed:
Testing processing just the first page of pdfs for thumbnails and ser… (#216)

* Testing processing just the first page of pdfs for thumbnails and service files

* Update Dockerfile

nigelgbanks committed:
Allow arguments to be passed to Solr on startup to configure memory, etc. (#253)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

dependabot[bot] committed:
Bump minimist in /code-server/rootfs/usr/local/share/.config/yarn/global (#255)

Bumps [minimist](https://github.com/minimistjs/minimist) from 1.2.5 to 1.2.7.
- [Release notes](https://github.com/minimistjs/minimist/releases)
- [Changelog](https://github.com/minimistjs/minimist/blob/main/CHANGELOG.md)
- [Commits](https://github.com/minimistjs/minimist/compare/v1.2.5...v1.2.7)

---
updated-dependencies:
- dependency-name: minimist
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nigelgbanks committed:
Use args for versions, and checksums (#256)

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com:w>

rosiel committed:
Configure solr when islandora_defaults is not present. (#258)

DonRichards committed:
Solr 8.11.2 update (#244)

* Update Dockerfile

* Add info for switch Solr versions

* Add new line after header

* Switching out ENV for ARG

* Switching out ENV for ARG

* Add a newline

* Remove "Solr keys" ref

* Solr 8.11.2 requires GNU ps to start correctly.

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com>

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=111&src=https%3A%2F%2Fgithub.com%2Fjhu-idc%2Fidc-isle-buildkit">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/jhu-idc/idc-isle-buildkit/111/c7fcd530/8022d93ac30c1779722c2e618334157e6a4167b9.svg)](https://app.codesee.io/r/reviews?pr=111&src=https%3A%2F%2Fgithub.com%2Fjhu-idc%2Fidc-isle-buildkit)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

nigelgbanks committed:
Added hypercube fedora settings back in (#265)

dannylamb committed:
Update default fcrepo URL for Hypercube to use port 8080 (#266)

* Update README.md

* Update Dockerfile

dannylamb committed:
s/8008/8080 in hypercube/Dockerfile

nigelgbanks committed:
Fix for build error, upstream package changed it's checksum.

nigelgbanks committed:
Remove old secret and use new one.

nigelgbanks committed:
Remove dead code & Description action for removed images.

nigelgbanks committed:
Bumped update Dockerhub description action version

nigelgbanks committed:
Issue 215 update alpine (#238)

* Updating alpine to the latest

- Updated PHP to 8.1
- Updated Java to 17 LTS (To get around deadlocks: https://github.com/docker/for-mac/issues/5590)
- Updated Code Server to 4.8.3
- Updated Matomo to 4.12.3
- Updated to Alpaca 2.2.0
- Updated to s6-overlay 3.1.2.1
- Overhaul of s6-overlay to support the major upgrade
- Moved to Alpaca standalone jar
- Removed the demo image as it is unmaintained and doesn't support php 8.1
- Removed Karaf image as it is no longer needed.
- Created test image for testing changes to isle-buildkit (includes sample content)
- Added Automated integration tests for a full site
- Updated documentation

* Removed fcrepo 5, as we'll only be supporting fcrepo 6 going forward.

* Moved Gradle plugin into repo and fixed tests.

* Added back dependencies for ./gradlew up

* Global find and replace mishap

* Fixes for running local docker-compose.yml file for testing.

* Improve build times with Github Actions

Updated documentation.

* Changed crayfish to pull from non-fork, as fix was merged.

* Removed recast

* Tesseract can now extract text from jp2 images.

* Added documentation for IDE/PHPStorm.

* Updated README to make pre-commit a explicit requirement.

* Added Cantaloupe to the list of services displayed when doing `make up`

* Support for generating certificates on Windows for Edge & Chrome.

nigelgbanks committed:
Fix for updating description action.

nigelgbanks committed:
Update fcrepo to latest release and fix default environment vars for alpaca.

nigelgbanks committed:
Roll back update to fcrepo6 as it causes tests to fail since it no longer respects SIGTERM.

nigelgbanks committed:
Try to make Fcrepo shutdown successfully...

nigelgbanks committed:
Need to specify a default user for dockerhub

nigelgbanks committed:
Disable persistent session in fcrepo container as it prevents shutdown

jefferya committed:
Missing icons in Matomo install (#275)

* addresses missing icons in Matomo https://github.com/Islandora-Devops/isle-buildkit/issues/274

* uses advised download location https://github.com/matomo-org/matomo/issues/16553 that contains the Git Submodule contents

* Added back missing composer files.

---------

Co-authored-by: Nigel Banks <nigel.g.banks@gmail.com>

nigelgbanks committed:
Better support for bind mounts with Drupal/IDE for development environments.

nigelgbanks committed:
Updated crayfits composer.lock against vunerability, and updated docs that describe this process.

nigelgbanks committed:
Fix predicate for checking if grype database needs updating

nigelgbanks committed:
Updated code-server extensions to latest

nigelgbanks committed:
Catch failures like ld-linux-aarch64.so.1: /opt/code-server/lib/node: Not a valid dynamic program while building.

Commit:
Do no use the packaged node executable as it fails to work on arm64, instead use the system node

Commit:
Fix cross platform builds of code-server, previously worked on arm64 only if built on an arm64 host.

nigelgbanks committed:
Merge extenstion installation into single run command.

nigelgbanks committed:
Installing concurrently can cause issues so we do it sequentially.

nigelgbanks committed:
Derp worked locally cause folders, need to be files to get checked in.

nigelgbanks committed:
Moved builder setup to isle-builder repository.

nigelgbanks committed:
Fixed capitalization function

nigelgbanks committed:
Cleanup matomo environment urls to split http/https

nigelgbanks committed:
Update starter site

nigelgbanks committed:
Support for changes in layout for the starter site in the test image.

nigelgbanks committed:
Need to bump the number of nodes to check for to account for new collection

nigelgbanks committed:
Run development-environment before confd as the file ownership needs to be set prior to render if bind mounting

nigelgbanks committed:
Little speed up on restart by not runing chown again.

nigelgbanks committed:
Check the actual folder that typically gets bind mounted.

nigelgbanks committed:
Allow drupal container to access broker via REST API

nigelgbanks committed:
Fixed race condition for changing folder permissions on startup in the ide.

nigelgbanks committed:
Added missing gif for PHPStorm setup

nigelgbanks committed:
Add Glibc for PHPStorm remote connections.

nigelgbanks committed:
issue-280 Fix for IP Address forwarding (#281)

nigelgbanks committed:
Update alpine/drush launcher. (#285)

* Upgrade Alpine to latest release

* Upgrade drush launcher

nigelgbanks committed:
Drush 12 is no longer compatiable with Drush launcher. (#286)

noahwsmith committed:
upgrade to fcrepo 6.4.0 (#288)

noahwsmith committed:
Docs should say FCREPO 6.4.0

joecorall committed:
Drupal 10 support (#289)

* Remove pipe flag from drush

* Also fix reverse_proxy_trusted_headers

joecorall committed:
Drupal 10 uses solr 8 (#290)

nigelgbanks committed:
Fixed find and replace issue.

nigelgbanks committed:
Fedora 5 support has been dropped in the 2.0 release, default to FEDORA6, though keep the option for those unable to upgrade to Fedora6 for some reason, though likely they will not beable to use the 2.0 release anyways.

nigelgbanks committed:
Added test for fcrepo6 file backend.

joecorall committed:
Add english OCR training data (#292)
